src/store/modules/imageUploaderStore.js

```diff
@@ -0,0 +1,64 @@
+import axios from 'axios';
+const aws = require('aws-sdk')
+
+
+export const imageUploaderStore = {
+    mutations: {
+        SET_IMAGEPATH(state, newState) {
+            this.state.imagePath = newState.data.imagePath
+        }
+    },
+
+    actions: {
+        upload_to_s3(context, payload) {
+            const file = payload;
+            aws.config.update({
+                secretAccessKey: process.env.DEV_SECRET_ACCESS_KEY,
+                accessKeyId: process.env.DEV_ACCESS_KEY_ID,
+            })
+            const s3 = new aws.S3({
+                signatureVersion: 'v4',
+                region: process.env.DEV_AWS_REGION,
+            })
+            
+
+            const key = 'static/uploads/' + Date.now() + '-' + file.name.replace(/\s/g, '-')
```

Comment:
I dont think the key i.e fileName needs to have `static/uploads` as a prefix. Why did you add it?

Reply:
if you are uploading a file to filesystem, then thats the path where the file will live but you are uploading directly to S3 here.